cvat/apps/engine/views.py

```diff
@@ -61,809 +79,1122 @@
+				required=False, enum=['download'], type=openapi.TYPE_STRING)],
+		responses={'202': openapi.Response(description='Dump of annotations has been started'),
+			'201': openapi.Response(description='Annotations file is ready to download'),
+			'200': openapi.Response(description='Download of file started')})
+	@action(detail=True, methods=['GET'], serializer_class=None,
+		url_path='annotations/(?P<filename>[^/]+)')
+	def dump(self, request, pk, filename):
+		"""
+		Dump of annotations in common case is a long process which cannot be performed within one request.
+		First request starts dumping process. When the file is ready (code 201) you can get it with query parameter action=download.
+		"""
+		filename = re.sub(r'[\\/*?:"<>|]', '_', filename)
+		username = request.user.username
+		db_task = self.get_object() # call check_object_permissions as well
+		timestamp = datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
+		action = request.query_params.get("action")
+		if action not in [None, "download"]:
+			raise serializers.ValidationError(
+				"Please specify a correct 'action' for the request")
+
+		dump_format = request.query_params.get("format", "")
+		try:
+			db_dumper = AnnotationDumper.objects.get(display_name=dump_format)
+		except ObjectDoesNotExist:
+			raise serializers.ValidationError(
+				"Please specify a correct 'format' parameter for the request")
+
+		file_path = os.path.join(db_task.get_task_artifacts_dirname(),
+			"{}.{}.{}.{}".format(filename, username, timestamp, db_dumper.format.lower()))
+
+		queue = django_rq.get_queue("default")
+		rq_id = "{}@/api/v1/tasks/{}/annotations/{}/{}".format(username, pk, dump_format, filename)
+		rq_job = queue.fetch_job(rq_id)
+
+		if rq_job:
+			if rq_job.is_finished:
+				if not rq_job.meta.get("download"):
+					if action == "download":
+						rq_job.meta[action] = True
+						rq_job.save_meta()
+						return sendfile(request, rq_job.meta["file_path"], attachment=True,
+							attachment_filename="{}.{}".format(filename, db_dumper.format.lower()))
+					else:
+						return Response(status=status.HTTP_201_CREATED)
+				else: # Remove the old dump file
+					try:
+						os.remove(rq_job.meta["file_path"])
+					except OSError:
+						pass
+					finally:
+						rq_job.delete()
+			elif rq_job.is_failed:
+				exc_info = str(rq_job.exc_info)
+				rq_job.delete()
+				return Response(data=exc_info, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+			else:
+				return Response(status=status.HTTP_202_ACCEPTED)
+
+		rq_job = queue.enqueue_call(
+			func=annotation.dump_task_data,
+			args=(pk, request.user, file_path, db_dumper,
+				  request.scheme, request.get_host()),
+			job_id=rq_id,
+		)
+		rq_job.meta["file_path"] = file_path
+		rq_job.save_meta()
+
+		return Response(status=status.HTTP_202_ACCEPTED)
+
+	@action(detail=True, methods=['POST'], serializer_class=None, url_path="get_object_counts")
+	def get_object_counts(self, request, pk):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5609236)